core/controllers/collection_viewer.py

```diff
@@ -80,23 +80,20 @@ def get(self, collection_id):
 class CollectionDataHandler(base.BaseHandler):
     """Provides the data for a single collection."""
 
+    @require_collection_playable
     def get(self, collection_id):
         """Populates the data on the individual collection page."""
-        allow_invalid_explorations = bool(
-            self.request.get('allow_invalid_explorations'))
 
         try:
             collection_dict = (
                 summary_services.get_learner_collection_dict_by_id(
                     collection_id, self.user_id,
-                    allow_invalid_explorations=allow_invalid_explorations))
+                    allow_invalid_explorations=False))
         except Exception as e:
             raise self.PageNotFoundException(e)
 
         self.values.update({
-            'can_edit': (
-                self.user_id and rights_manager.Actor(self.user_id).can_edit(
-                    feconf.ACTIVITY_TYPE_COLLECTION, collection_id)),
+            'can_edit': False,
```

Comment:
I did not even consider that the can_edit attribute was used for a variety of things, will change back.